inference-engine/src/mkldnn_plugin/nodes/depth_to_space.cpp

```diff
@@ -8,113 +8,210 @@
-            own_dims[4] = block_size;
-
-            size_t C = src_dims[src_dims.size() - 2] * src_dims[src_dims.size() - 1];
-            ownStrides[0] = src_dims[src_dims.size() - 3] * C;
-            ownStrides[1] = src_dims[src_dims.size() - 1];
-            ownStrides[2] = block_size * C;
-            ownStrides[3] = 1;
-            ownStrides[4] = C;
-            work_amount_dst = ownStrides[0] * own_dims[0];
-
-            addConfig(layer, { DataConfigurator(ConfLayout::PLN) }, { DataConfigurator(ConfLayout::PLN) });
+                THROW_IE_EXCEPTION << "DepthToSpace layer with name '" << layer->name << "' has incorrect number of input/output edges";
+
+            inDims = layer->insData[0].lock()->getTensorDesc().getDims();
+            if (inDims.size() < 3)
+                THROW_IE_EXCEPTION << "DepthToSpace layer with name '" << layer->name << "' has incorrect number of input dimensions";
+
+            if (inDims.size() > 5)
+                THROW_IE_EXCEPTION << "DepthToSpace layer with name '" << layer->name << "' doesn't support dimensions with rank greater than 5";
+
+            SizeVector outDims = layer->outData[0]->getTensorDesc().getDims();
+            if (inDims.size() != outDims.size())
+                THROW_IE_EXCEPTION << "DepthToSpace layer with name '" << layer->name << "' has incorrect number of input/output dimensions";
+
+            std::string modeString = layer->GetParamAsString("mode");
+            if (modeString == "blocks_first") {
+                mode = DepthToSpaceMode::BLOCKS_FIRST;
+            } else if (modeString == "depth_first") {
+                mode = DepthToSpaceMode::DEPTH_FIRST;
+            } else {
+                THROW_IE_EXCEPTION << "DepthToSpace layer with name '" << layer->name << "' doesn't support mode: " << modeString;
+            }
+
+            blockSize = layer->GetParamAsUInt("block_size", 1);
+            if (blockSize == 0)
+                THROW_IE_EXCEPTION << layer->name << " Incorrect blockSize parameter is zero!";
+
+            size_t numSpatialDims = inDims.size() - 2;
+            blockStep = static_cast<size_t>(std::pow(blockSize, numSpatialDims));
+            if (inDims[1] % blockStep)
+                THROW_IE_EXCEPTION << "DepthToSpace layer with name '" << layer->name <<
+                    "' has block_size parameter which is incompatible with input tensor channels dimension size";
+
+            if (inDims[1] / blockStep != outDims[1])
+                THROW_IE_EXCEPTION << "DepthToSpace layer with name '" << layer->name << " has incompatible input/output channels";
+
+            for (int i = 0; i < numSpatialDims; i++) {
+                if (inDims[i + 2] != outDims[i + 2] / blockSize)
+                    THROW_IE_EXCEPTION << "DepthToSpace layer with name '" << layer->name << " has incompatible spatial dims";
+            }
+
+            auto computePrc = layer->insData[0].lock()->getTensorDesc().getPrecision();
```

Comment:
It might be the situation when we have different input and output precision (e.g. because of BF16 marking). So we just uses equal input/output precision inside the node. Plug-in will insert reorder after it automatically.

Reply:
Discussed with @dmitry-gorokhov. That is a feature.
In case of original CNNLayer has different precisions on input/output we will select the input as execution precision.  Precision/layout solving mechanic will cover that case and inject corresponding reorder after this layer.